use-cases/client-server-msgs-perf/svmsg_file_client.c

```diff
@@ -61,44 +92,99 @@ main(int argc, char *argv[])
         errExit("atexit");
     }
 
-    /* Send message asking for file named in argv[1] */
+    size_t niters = atoi(argv[1]);
+    printf("Client: ID=%d Perform %lu message-exchanges to increment a number"
+#if L3_ENABLED
+           ", with L3-logging enabled on server-side"
+#endif // L3_ENABLED
+           ".\n", clientId, niters);
 
-    req.mtype = 1;                      /* Any type will do */
-    req.clientId = clientId;
-    strncpy(req.pathname, argv[1], sizeof(req.pathname) - 1);
-    req.pathname[sizeof(req.pathname) - 1] = '\0';
-                                        /* Ensure string is terminated */
+    requestMsg req;
+    req.mtype       = REQ_MT_INIT;     // New client is establishing a connection
+    req.clientId    = clientId;
+    req.client_idx  = REQ_CLIENT_INDEX_UNKNOWN;
+    req.counter     = 0;
 
     if (msgsnd(serverId, &req, REQ_MSG_SIZE, 0) == -1) {
         errExit("msgsnd");
     }
 
-    /* Get first response, which may be failure notification */
-
+    responseMsg resp;
     msgLen = msgrcv(clientId, &resp, RESP_MSG_SIZE, 0, 0);
     if (msgLen == -1) {
         errExit("msgrcv");
     }
 
     if (resp.mtype == RESP_MT_FAILURE) {
-        printf("%s\n", resp.data);      /* Display msg from server */
+        /* Display msg from server */
+        printf("Counter=%" PRIu64 "\n", resp.counter);
         exit(EXIT_FAILURE);
     }
 
-    /* File was opened successfully by server; process messages
-       (including the one already received) containing file data
-    */
-    totBytes = msgLen;                  /* Count first message */
-    for (numMsgs = 1; resp.mtype == RESP_MT_DATA; numMsgs++) {
-        msgLen = msgrcv(clientId, &resp, RESP_MSG_SIZE, 0, 0);
+    // Re-establish client's identity as obtained from the server.
+    req.client_idx = resp.client_idx;
+
+    /* Loop around for requested # of iterations, or till the time the
+     * server informs us to quit. (We implement receiving REQ_MT_QUIT from
+     * the server, but that is not happening right now.)
+     */
+    size_t ictr;
+
+    // Msg-round trip timing measurements for each msgsend/recv() pair.
+    struct timespec ts0;
+    struct timespec ts1;
+    uint64_t        elapsed_ns = 0;
+
+    for (ictr = 0; ictr < niters; ictr++) {
+
+        if ((ictr % L3_100K) == 0) {
+            printf(".");
+            fflush(stdout);
+        }
+        req.counter = resp.counter;
+        req.mtype = REQ_MT_INCR;
+
+        if (clock_gettime(CLOCK_REALTIME, &ts0)) {              // Timing begins
+            errExit("clock_gettime-ts0");
+        }
+
+        if (msgsnd(serverId, &req, REQ_MSG_SIZE, 0) == -1) {    // --> Send
+            errExit("msgsnd");
+        }
+
+        msgLen = msgrcv(clientId, &resp, RESP_MSG_SIZE, 0, 0);  // <-- Recv
```

Comment:
Any L3-logging enabled on the server-side will be included in this elapsed-time region.